src/gluonnlp/embedding/token_embedding.py

```diff
@@ -1020,9 +1020,85 @@ class Word2Vec(TokenEmbedding):
 
     def __init__(self, source='GoogleNews-vectors-negative300',
                  embedding_root=os.path.join(get_home_dir(), 'embedding'), **kwargs):
-        self._check_source(self.source_file_hash, source)
-
         super(Word2Vec, self).__init__(**kwargs)
-        pretrained_file_path = self._get_file_path(self.source_file_hash, embedding_root, source)
+        if source:
+            self._check_source(self.source_file_hash, source)
+            pretrained_file_path = self._get_file_path(self.source_file_hash, embedding_root,
+                                                       source)
+            self._load_embedding(pretrained_file_path, elem_delim=' ')
 
-        self._load_embedding(pretrained_file_path, elem_delim=' ')
+    def load_w2v_binary(self, pretrained_file_path, encoding='utf8'):
+        """Load embedding vectors from a binary pre-trained token embedding file.
+
+        Parameters
+        ----------
+        pretrained_file_path: str
+            The path to a binary pre-trained token embedding file.
+        encoding: str
+            The encoding type of the file.
```

Comment:
Why does the binary format have a encoding parameter?

Reply:
It helps to read word2vec from other languages, e.g. Chinese. For example, all Chinese chars/tokens can be used in unicode(utf8) by passing encoding param. (Actually, utf-8 is the default value already)